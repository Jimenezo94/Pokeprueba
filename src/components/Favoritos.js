import {
  collection,
  deleteDoc,
  doc,
  getDocs,
  updateDoc,
} from "firebase/firestore";
import React, { useEffect, useState } from "react";
import { db } from "../firebaseConfig";
import { Modal } from "react-bootstrap";

const Favoritos = () => {
  const [objtFav, setObjfav] = useState([]);
  const [favoritos, setFavoritos] = useState([]);
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const listitem = async () => {
    let arrayfav = [];

    const querySnapshot = await getDocs(collection(db, "favoritos"));
    console.log(querySnapshot);

    querySnapshot.forEach((doc) => {
      // console.log(doc.id);
      arrayfav.push({
        id: doc.id,
        ...doc.data(),
      });
    });
    console.log(favoritos);
    setFavoritos(arrayfav);
  };

  useEffect(() => {
    listitem();
    console.log("jj", favoritos);
  }, []);

  const Editfav = async (id, data) => {
    const getitemfav = doc(db, "favoritos", id);
    await updateDoc(getitemfav, data);
  };
  
  const deletefav = (id) => {
    console.log("ggg", id);
    deleteDoc(doc(db, "favoritos", id));
    listitem();
  };

  const onSubmit = (e) => {
    e.preventDefault();
    console.log(objtFav);
    Editfav(objtFav.id, objtFav);
    listitem();
  };


  const handleChange = (event) => {
    setObjfav({
      ...objtFav,
      [event.target.name]: event.target.value,
    });
  };



  return (
    <div>
      <table className="table text-center mt-3">
        <thead>
          <tr>
            <th scope="col">Username</th>
            <th scope="col">Nombre</th>
            <th scope="col">Description</th>
            <th scope="col">Action</th>
          </tr>
        </thead>
        <tbody>
          {favoritos.map((poke, id) => (
            <tr key={id}>
              {/* <td><img src={e.url} width="50" height="50" alt=""/></td> */}
              <td>{poke.altura}</td>
              <td>{poke.name}</td>
              <td>{poke.peso}</td>
              <td>
                <input
                  value="Delete"
                  type="button"
                  className="btn btn-outline-dark"
                  onClick={() => {
                    deletefav(poke.id);
                  }}
                />
              </td>
              <td>
                <input
                  value="Edit"
                  type="button"
                  className="btn btn-outline-dark"
                  onClick={() => {
                    handleShow();
                    setObjfav(poke);
                  }}
                />
              </td>
            </tr>
          ))}
        </tbody>
      </table>
      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Modal heading</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <form onSubmit={(e) => onSubmit(e)}>
            <input
              type="text"
              className="form-control mt-2"
              name="name"
              autoComplete="off"
              placeholder="nombre"
              value={objtFav.name}
              onChange={handleChange}
              required
            />

            <input
              type="text"
              className="form-control mt-2"
              name="altura"
              autoComplete="off"
              placeholder="email"
              value={objtFav.altura}
              onChange={handleChange}
              required
            />

            <textarea
              className="form-control mt-2"
              autoComplete="off"
              name="peso"
              value={objtFav.peso}
              placeholder="peso"
              onChange={handleChange}
              required
            ></textarea>

            <div className="d-grid gap-2 mx-auto mt-2">
              <input
                value="Save"
                type="submit"
                className="btn btn-outline-dark"
              />
            </div>
          </form>
        </Modal.Body>
      </Modal>
    </div>
  );
};

export default Favoritos;
