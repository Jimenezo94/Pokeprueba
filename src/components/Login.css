.login {
    background-color: white;
    height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
  }
  
  .login > p {
    margin-top: 10px;
    color: darkgray;
  }
  
  .login-logo {
    margin-top: 20px;
    margin-bottom: 20px;
    object-fit: contain;
    width: 100px;
    margin-right: auto;
    margin-left: auto;
  }
  
  .login-container {
    width: 350px;
    height: fit-content;
    display: flex;
    flex-direction: column;
    border-radius: 5px;
    padding: 60px;
  }
  
  .login-container > h1 {
    font-weight: 500;
    margin-bottom: 20px;
  }
  
  .login-container > form > h5 {
    font-size: 14px;
    margin-bottom: 5px;
  }
  
  .login-container > form > input {
    height: 30px;
    margin-bottom: 10px;
    background-color: #fff;
    width: 98%;
  }
  
  .login-container > p {
    margin-top: 15px;
    font-size: 12px;
  }
  
  .login-signIn{
    background: #cc73f0;
    border-radius: 2px;
    width: 100%;
    height: 30px;
    border: 1px solid;
    margin-top: 10px;
    color: #111;
    cursor: pointer;
  }
  
  .login-register {
    border-radius: 2px;
    width: 100px;
    height: 30px;
    border: 1px solid;
    margin:20px  10px;
    border-color: darkgray;
    font-size: 12.5px;
    cursor: pointer;
  }