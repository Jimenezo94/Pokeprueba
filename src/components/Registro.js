import React, { useState, useEffect } from "react";
import './registro.css'
//import { Form, Button, Container } from "react-bootstrap";
import { NavLink, useNavigate } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { loginEmailPassword, loginGoogle } from "../actions/actionLogin";
import { RegisterInitiate } from "../actions/actionRegister";
import { POKE_KEY } from '../reducers/pokemonReducer';


function Registro() {

  const [email, setEmail] = useState('');
  const [password, setPassword]= useState('');

 // const {user} =useSelector((state) =>state.data)
  // console.log("state", user)
  // const history = useNavigate()
  const dispatch = useDispatch();

  let estados = useSelector((state) => {
    return state[POKE_KEY];
  
  });
    


  const registro = (e)=>{

    e.preventDefault();
    dispatch(RegisterInitiate(email,password))
    console.log(email,password)
    setEmail("")
    setPassword("")
  }
  
  useEffect(()=>{
    console.log("fsdf")
    if(estados.usuario){
    //history.push("/")
    console.log("history")
    }
      },[estados.usuario, dispatch])
      
  // const handleLogin = (e) => {
  //    e.preventDefault();
  //    dispatch(loginEmailPassword(email,password));
  // }

  // const handleGoogle = () => {
  //      dispatch(loginGoogle());
  // }

  return (
    <div className="register">
      <NavLink to="/">
        <img src="" alt="logo" className="register-logo" />
      </NavLink>
      <div className="register-container">
        <h1>Registro</h1>
        <form action="">
          <h5>E-mail</h5>
          <input
            type="text"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
           <h5>Password</h5>
          <input
            type="text"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
          <button type='submit' 
          className='login-signIn'
          onClick={registro}>
              Registrate
          </button>
        </form>
        <p>Ya tienes una cuenta?</p>
        <NavLink to="/">
            <button className='sigIn-link'>
               Inicia sesion
            </button>



        </NavLink>
      </div>
    </div>
  );
}

export default Registro;


