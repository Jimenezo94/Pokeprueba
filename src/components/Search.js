import React from 'react'
import {useSelector, useDispatch } from 'react-redux'
import { actualizar, searchPokemon } from '../actions/pokeactions'
import { POKE_KEY } from '../reducers/pokemonReducer';

const SearchBar = () => {

    let dispatch = useDispatch () 

    const handleInpuSearch= (e) => {
        dispatch(searchPokemon(e))
        console.log(e, estados.search, 'ttt')
        
        let datos= estados.data.results.filter(function(element){
            if(element.name.startsWith(e)){
                return element;

            }
          });
         
          let obj ={ 
              results:datos
          }
           if(estados.search.length === 0){
            console.log('lon')  
            obj = estados.data
          }
          //estados.data_filtrada= obj
          dispatch(actualizar(obj))
        console.log( estados.data_filtrada, 'data',obj)
    }
    let estados = useSelector((state) => {
    return state[POKE_KEY];
    
  });

    const click_search= (e) => {
        e.preventDefault();
        //dispatch(searchPokemon(e))
        console.log( estados.search, 'hfhfghfg')
        console.log( estados.search.length, 'hfhfghfg')
        //console.log( estados.data.results.find( fruta => fruta.name === 'raichu' ), 'data.results')

        let datos= estados.data.results.filter(function(element){
            return element.name === estados.search;
          });
         
          let obj ={ 
              results:datos
          }
           if(estados.search.length === 0){
            console.log('lon')  
            obj = estados.data
          }
          //estados.data_filtrada= obj
          dispatch(actualizar(obj))
        console.log( estados.data_filtrada, 'data',obj)
    }
    

  return (
   
  <div className='Searcher wrapper'>

             
<form htmlFor='for'>
  <label for="gsearch">Busca tu pokemon favorito: </label>
  <input type="search" 
  id="gsearch" 
  name="gsearch"
  onChange={event => handleInpuSearch(event.target.value)}/>
  <input type="submit" onClick={event =>click_search(event)}/> 
</form>
       
    
          </div>

  )
}

export default SearchBar;