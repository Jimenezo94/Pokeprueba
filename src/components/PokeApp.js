import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { db } from "../firebaseConfig";

import { loadData, loadPokeName } from "../actions/pokeactions";
import { POKE_KEY } from "../reducers/pokemonReducer";
import SearchBar from "./Search";
import Favoritos from "./Favoritos";
import axios from "axios";
import { addDoc, collection } from "@firebase/firestore";

const PokeApp = () => {
  let dispatch = useDispatch();
  // view store
  let estados = useSelector((state) => {
    return state[POKE_KEY];
  });

  const activateFavs = async (p) => {
    console.log("jjffgwe", p);
    dispatch(loadPokeName(p.name));
    //  setTimeout(()=>{console.log(estados.pokeUrlData)},3000)

    // dispatch(addTofav(p.items))
    try {
      console.log("I am back here: ", estados.name);
      let dataURL = `https://pokeapi.co/api/v2/pokemon/${p.name}`;
      let response = await axios.get(dataURL);
      console.log(response);
      let objadd = {
        name: p.name,
        altura: response.data.height,
        peso: response.data.weight,
        tipo: response.data.types,
        usuario: "a1@a.com",
      };
      addDoc(collection(db, "favoritos"), objadd)
        .then((resp) => {
          console.log(objadd);
          //window.location.href = '/fav'
        })
        .catch((error) => {
          console.log(error);
        });
    } catch (error) {
      console.log("EEER", error);
    }
  }; 

  useEffect(() => {
    dispatch(loadData());
    console.log("estados.", estados.data_filtrada);
  }, []);

  console.log(POKE_KEY, "kkkj");

  return (
    <div>
      <div className="card">
        <Link to="/fav">
          <p>Favoritos</p>
        </Link>{" "}
        <div className="card-header centered">
          <h5>THE POKE APP</h5>
        </div>
        <SearchBar />
        <div className="card-body">
          <div className="pokeContainer">
            {estados.data_filtrada.length === 0 ? null : (
              <React.Fragment>
                {estados.data_filtrada.results.map((item) => {
                  return (
                    <div key={item.name}>
                      <div className="pokeItem">
                        <div className="pokeNameContainer centered">
                          <p>{item.name}</p>
                        </div>

                        <div className="pokeNameContainer centered"></div>
                        <div className="viewImageContainer centered">
                          <Link to={`/details/${item.name}`}>
                            <p>Detalles</p>
                          </Link>
                        </div>
                      </div>
                      {/* <button
                         onClick={additem()}
                         >Add
                        </button> */}

                      <button style={{margin:'auto'}}onClick={() => activateFavs(item)}>
                        Fav
                      </button>
                    </div>
                  );
                })}
              </React.Fragment>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default PokeApp;
