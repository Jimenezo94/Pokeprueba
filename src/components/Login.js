import React, { useState, useEffect } from "react";
//import { Form, Button, Container } from "react-bootstrap";
import { NavLink} from "react-router-dom";
import './Login.css'
import { loginEmailPassword, loginGoogle } from "../actions/actionLogin";
import { useDispatch, useSelector } from "react-redux";
import { POKE_KEY } from '../reducers/pokemonReducer';

function Login() {
 // const dispatch = useDispatch();

  const [email, setEmail] = useState('');
  const [password, setPassword]= useState('');
  const dispatch = useDispatch();
  const singIn = (e)=>{

    e.preventDefault();
    dispatch(loginEmailPassword(email,password))
    setEmail("")
    setPassword("")
  }
  // const singInGoogle = (e)=>{

  //   e.preventDefault();
  //   dispatch(loginGoogle())
  //   setEmail("")
  //   setPassword("")
  // }
  let estados = useSelector((state) => {
    return state[POKE_KEY];
  
  });

  const handleGoogle = () => {
       dispatch(loginGoogle());
  }
  useEffect(()=>{

    if(estados.user){
    //history.push("/")
    console.log("history")
    }
      },[estados.user, dispatch])
  return (
    <div className="login">
     
      <div className="login-container">
        <h1>Sign In</h1>
        <form action="">
          <h5>E-mail</h5>
          <input
            type="text"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
           <h5>Password</h5>
          <input
            type="text"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
          <button type='submit' 
          className='login-signIn'
          onClick={singIn}>
              Inicia sesi√≥n
          </button>
        </form>
        <button type='' 
          className='login-signIn'
          onClick={handleGoogle}>
              Inicia con google
          </button>
        <NavLink to="/register">


            <button className='login-register'>
              
                Crea una cuenta
            </button>



        </NavLink>
      </div>
    </div>
  );
}

export default Login;

