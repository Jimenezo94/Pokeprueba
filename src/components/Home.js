import {Provider } from 'react-redux'
import React from 'react'
import PokeApp from './PokeApp'
import store from "../store";
// import PokeDetails from './PokeDetails'

const Home = () => {
    return (
        <React.Fragment>
            
            <div>
                {/* <PokeDetails/> */}
             
                <PokeApp/>
             
            </div>
        </React.Fragment>
      
    )
}

export default Home