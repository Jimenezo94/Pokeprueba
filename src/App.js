import Login from "./components/Login";
import Registro from "./components/Registro";
import "./App.css";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Home from "./components/Home";
import PokeDetails from "./components/PokeDetails";


function App() {
  return (
  <BrowserRouter > 

    <Routes>

      <Route path='/' exact element={<Login/>}/>
      <Route path='/register' exact element={<Registro/>}/> 
    <Route path='/home' exact element={<Home/>}/>
    <Route path='/details/:url'exact element={<PokeDetails/>}/>

    </Routes>
    
    </BrowserRouter>
  );
}

export default App;
