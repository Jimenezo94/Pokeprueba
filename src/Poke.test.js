import React from 'react';
import { shallow} from 'enzyme';
import * as types from './types';
import App from './App';
import pokeReducer from './reducers/pokemonReducer';
import Home from './components/Home';
import Favoritos from './components/Favoritos';
import Registro from './components/Registro';
import Search from './components/Search';
import Login from './components/Login';
import PokeApp from './components/PokeApp';
import { mount } from 'enzyme/build';
import configureStore from 'redux-mock-store';
import { render, screen } from '@testing-library/react';
import {Provider } from 'react-redux'
import store from "./store";
import { Button } from 'bootstrap';

describe('pruebas', ()=>{
    const initialState = { output: 10 };
    const mockStore = configureStore();

    test('debería mostrarse', () => {
      const wrapper = shallow(<Home/>);
  
      expect(wrapper).toMatchSnapshot();
  
   
    })
  
   
    test('estado por defecto', ()=>{
        const state = pokeReducer({loading:true}, {})
  
        expect (state).toEqual({loading:true});
  
    })
    // test('debería regi', () => {
    //   const wrapper = shallow(<PokeApp/>);
  
    //   expect(wrapper).toMatchSnapshot();
  
   
    // })
    // test('debería search', () => {
    //   const wrapper = shallow(<SearchBar/>);
  
    //   expect(wrapper).toMatchSnapshot();
  
   
    // })
  
  
    test('user es', ()=>{
      const state = pokeReducer({user:null}, {})
  
      expect (state).toEqual({user:null});
  })
  test('obj', ()=>{
    expect(types).toEqual({
      ACTUALIZAR_DATA: "ACTUALIZAR_DATA",
         LOAD_DATA_FAILURE: "LOAD_DATA_FAILURE",
         LOAD_DATA_REQUEST: "LOAD_DATA_REQUEST",
         LOAD_DATA_SUCCESS: "LOAD_DATA_SUCCESS",
         LOAD_DATA_SUCCESS_2: "LOAD_DATA_SUCCESS_2",
         LOAD_POKE_FAILURE: "LOAD_POKE_FAILURE",
         LOAD_POKE_REQUEST: "LOAD_POKE_REQUEST",
         LOAD_POKE_SUCCESS: "LOAD_POKE_SUCCESS",
         SEARCH: "SEARCH",
         addfav: "addfav",
          login: "login",
         login_fallido: "login_fallido",
         login_start: "login_start",
         login_success: "login_success",
         logout_fallido: "logout_fallido",
         logout_start: "logout_start",
         logout_success: "logout_success",
         register: "register",
         register_fallido: "register_fallido",
            register_start: "register_start",
      set_user: "set_user",
     })
  })
  
  
  test('user',()=>{
      const loginE = {
          type:types.login,
          payload: {
            email:'',
            password:'',
         }
       }
  
  const state = pokeReducer ({loading:false }, loginE)
  expect(state).toEqual({
    loading: false, 
    user:{
      email:'', password:'',
   }
  })
        
  })
  
  test('estado favoritos',()=>{
    const action = {
        type:types.SEARCH,
        payload: 'word'
     }

const state = pokeReducer ({ }, action)
expect(state).toEqual({
  search:'word'
})
      
})
// it('should render correctly with no props', () => {
//   const component = shallow(<Home />);
  
//   expect(component).toMatchSnapshot();
// });
// it('should disable submit button on submit click', () => {
//   //const wrapper = shallow(<PokeApp />);
//   const { wrapper } = render(
//     <Provider store={store}>
//         <App />
//     </Provider>
// );

//   // const submitButton = wrapper.find();
//   // submitButton.simulate('click');

//   // expect(submitButton.prop('disabled')).toBeTruthy();
//   //wrapper.instance().validateForm()
//   //expect(wrapper.html()).toMatchSnapshot();
//   expect(wrapper.find('Email').length).toEqual(1);
//  })
    // let store;

    it('Shows "Sign In!"', () => {
      //store = mockStore(initialState);
      const { getByText } = render(
          <Provider store={store}>
              <App />
          </Provider>
      );

      expect(screen.getByText('Sign In')).not.toBeNull();
  })
  it('Shows "earch"', () => {
    //store = mockStore(initialState);
    const { getByText } = render(
        <Provider store={store}>
            <Search />
        </Provider>
    );

    expect(screen.getByText('Busca tu pokemon favorito:')).not.toBeNull();
})
  })