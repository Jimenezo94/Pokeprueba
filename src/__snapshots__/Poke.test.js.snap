// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas deber√≠a mostrarse 1`] = `
<Fragment>
  <div>
    <PokeApp />
  </div>
</Fragment>
`;
